File: C++ for Everyone, 2nd Ed, Horstmann/Chapter2/P2-9.cpp
//Improve the program discussed in the How To 2.1 to allow input of quarters in
//addition to bills.

#include <iostream>
#include <iomanip>

using namespace std;
int main()
{
   const int PENNIES_PER_DOLLAR = 100;
   const int PENNIES_PER_QUARTER = 25;

   cout << "Enter value "
      << "(0.25 = quarter, 1.25 = $1.25, 5 = $5 bill, etc.): ";
   double cash_value;
   cin >> cash_value;

   cout << "Enter item price in pennies: ";
   int item_price;
   cin >> item_price;

   int amount_due = PENNIES_PER_DOLLAR * cash_value - item_price;
   int dollar_coins = amount_due / PENNIES_PER_DOLLAR;
   amount_due = amount_due % PENNIES_PER_DOLLAR;
   int quarters = amount_due / PENNIES_PER_QUARTER;
   cout << "Dollar coins: " << setw(6) << dollar_coins << endl
      << "Quarters:     " << setw(6) << quarters << endl;
   return 0;
}

File: Python For Everyone, Horstmann/Chapter2/P2.9.py
# Improve the program discussed in How To 2.1 to allow input of quar ters in addition
# to bills.

#constants
PENNIES_PER_DOLLAR = 100
PENNIES_PER_QUARTER = 25


cashValue = float(input("Enter value (0.25 = quarter, 1.25 ="
                        " $1.25, 5 = $5 bill, etc.): "))

itemPrice = int(input("Item price in pennies: "))

amountDue = PENNIES_PER_DOLLAR * cashValue - itemPrice
dollarCoins = amountDue / PENNIES_PER_DOLLAR

amountDue %= PENNIES_PER_DOLLAR
quarters = amountDue / PENNIES_PER_QUARTER

print("Dollar coins: %6.2f" % dollarCoins)
print("Quarters:     %6.2f" % quarters)